Charon/FileInterface.py

```diff
@@ -48,11 +46,11 @@ def flush(self) -> None:
         raise NotImplementedError("The flush() function of " + self.__class__.__qualname__ + " is not implemented.")
 
     ##  Returns a list of all resources and metadata in the file.
-    def listPaths(self) -> List[str]:
+    #   \param root_path The root path to search from if not the whole Zip file is required.
```

Comment:
I'm not sure if the addition of the root_path is a good thing. It complicates all subclass implementations, and can be solved with a filter on the result as well.

I suggest we:
- Remove this parameter
or:
- Keep this parameter, call a protected _listPaths that does the actual work, and have the filtering done in the base class. So filtering implementation doesn't have to be duplicated in all subclasses.

Reply:
Actually I'm gonna go for option 1 and do the filtering in my convenience methods.